primitives/npos-elections/src/lib.rs

```diff
@@ -479,60 +564,52 @@ impl<AccountId> StakedAssignment<AccountId> {
 ///
 /// This, at the current version, resembles the `Exposure` defined in the Staking pallet, yet they
 /// do not necessarily have to be the same.
-#[derive(Default, Debug)]
-#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Eq, PartialEq))]
+#[derive(Default, Debug, Encode, Decode, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
 pub struct Support<AccountId> {
 	/// Total support.
 	pub total: ExtendedBalance,
 	/// Support from voters.
 	pub voters: Vec<(AccountId, ExtendedBalance)>,
 }
 
-/// A linkage from a candidate and its [`Support`].
-pub type SupportMap<A> = BTreeMap<A, Support<A>>;
-
-/// Build the support map from the given election result. It maps a flat structure like
-///
-/// ```nocompile
-/// assignments: vec![
-///     voter1, vec![(candidate1, w11), (candidate2, w12)],
-///     voter2, vec![(candidate1, w21), (candidate2, w22)]
-/// ]
-/// ```
+/// A target-major representation of the the election outcome.
 ///
-/// into a mapping of candidates and their respective support:
+/// Essentially a flat variant of [`SupportMap`].
 ///
-/// ```nocompile
-///  SupportMap {
-///     candidate1: Support {
-///         own:0,
-///         total: w11 + w21,
-///         others: vec![(candidate1, w11), (candidate2, w21)]
-///     },
-///     candidate2: Support {
-///         own:0,
-///         total: w12 + w22,
-///         others: vec![(candidate1, w12), (candidate2, w22)]
-///     },
-/// }
-/// ```
+/// The main advantage of this is that it is encodable.
+pub type Supports<A> = Vec<(A, Support<A>)>;
+
+/// Linkage from a winner to their [`Support`].
 ///
-/// The second returned flag indicates the number of edges who didn't corresponded to an actual
-/// winner from the given winner set. A value in this place larger than 0 indicates a potentially
-/// faulty assignment.
+/// This is more helpful than a normal [`Supports`] as it allows faster error checking.
+pub type SupportMap<A> = BTreeMap<A, Support<A>>;
+
+/// Helper trait to convert from a support map to a flat support vector.
+pub trait FlattenSupportMap<A> {
```

Comment:
Good catch: I left it there because I expose both `to_support_map` and `to_support` and I thought it would be nice to expose an easy way to convert between the two as well (regardless of it being rather simple).

Reply:
I think ppl know how to convert a BTreeMap to a Vec but ok.